src/main/java/io/github/nucleuspowered/nucleus/modules/ticket/data/TicketDataManager.java

```diff
@@ -0,0 +1,294 @@
+                    } catch (SQLException ex) {
+                        ex.printStackTrace();
+                    }
+                });
+            } catch (SQLException ex) {
+                ex.printStackTrace();
+            }
+        });
+    }
+
+    public void deleteTicket(Ticket ticket) throws SQLException {
+        executor.execute(() -> {
+            try {
+                if (!getDataSource().isPresent())
+                    throw new SQLException("Deleting the Ticket in the database failed... the data source is not present.");
+                Connection connection = getDataSource().get().getConnection();
+                PreparedStatement statement = connection.prepareStatement(DELETE_TICKET_QUERY);
+
+                statement.setInt(1, ticket.getId());
+
+                //Check if anything was affected, if it wasn't throw an exception.
+                int rowsAffected = statement.executeUpdate();
+                if (rowsAffected == 0) {
+                    throw new SQLException("Deleting the Ticket in the database failed... no rows were affected.");
+                }
+
+                statement.close();
+                connection.close();
+            } catch (SQLException ex) {
+                ex.printStackTrace();
+            }
+        });
+    }
+
+    public void createTicketMessage(Ticket ticket, Instant creationDate, String message) throws SQLException {
+        executor.execute(() -> {
+            try {
+                if (!getDataSource().isPresent()) throw new SQLException("Creating the Ticket Message in the database failed... the data source is not present.");
+                Connection connection = getDataSource().get().getConnection();
+                PreparedStatement statement = connection.prepareStatement(CREATE_TICKET_MESSAGE_QUERY);
+
+                statement.setInt(1, ticket.getId());
+                statement.setTimestamp(2, Timestamp.from(creationDate));
+                statement.setString(3, message);
+
+                //Check if anything was affected, if it wasn't throw an exception.
+                int rowsAffected = statement.executeUpdate();
+                if (rowsAffected == 0) {
+                    throw new SQLException("Creating the Ticket Message in the database failed... no rows were affected.");
+                }
+
+                statement.close();
+                connection.close();
+            } catch (SQLException ex) {
+                ex.printStackTrace();
+            }
+        });
+    }
+
+    public CompletableFuture<TreeMap<Long, String>> retrieveTicketMessages(int ticketID) throws SQLException {
+        return cache.lookupTicketMessagesByTicketID(ticketID);
+    }
+
+    public void updateTicketMessage(Ticket ticket, Instant creationDate, String newMessage) throws SQLException {
+        executor.execute(() -> {
+            try {
+                if (!getDataSource().isPresent())
```

Comment:
Please add braces (`{}`) to all `if` statements - one line statements are less readable and can end up with some pretty servere bugs - such as [Apple's goto fail](https://www.imperialviolet.org/2014/02/22/applebug.html) bug.